Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m

```diff
@@ -143,14 +143,16 @@ - (UIView *)mediaView {
         NSURL *url =
             [MIMETypeUtil simLinkCorrectExtensionOfFile:_attachment.mediaURL ofMIMEType:_attachment.contentType];
 
-        AVURLAsset *asset         = [[AVURLAsset alloc] initWithURL:url options:nil];
-        _waveform                 = [[SCWaveformView alloc] init];
-        _waveform.frame           = CGRectMake(42.0, 0.0, size.width - 84, size.height);
-        _waveform.asset           = asset;
-        _waveform.progressColor   = [UIColor whiteColor];
-        _waveform.backgroundColor = [UIColor colorWithRed:229 / 255.0f green:228 / 255.0f blue:234 / 255.0f alpha:1.0f];
-        [_waveform generateWaveforms];
-        _waveform.progress = 0.0;
+        if (_waveform == NULL) {
+            AVURLAsset *asset         = [[AVURLAsset alloc] initWithURL:url options:nil];
+            _waveform                 = [[SCWaveformView alloc] init];
```

Comment:
I know this came before you, but prefer property access over direct ivar manipulation unless you are in an initializer. So all these `_waveform` should be replaced with `self.waveform`.